docs/design/flow-loop/flow-loop.md

```diff
@@ -0,0 +1,44 @@
+# Flow loop feature
+
+## Overview
+
+Flow loop feature designed for flow path testing. Loop provides additional flow rules on one of the terminating switch so any flow traffic is returned to switch-port where it was received. 
+Such flow has `looped=true` flag and support all flow operations. When loop removed system should restore original flow rules.
+
+![Flow-loop](flow-loop.png "Flow loop")
+
+## API
+
+* Get existing flow loops `GET /v2/flows/loop` with optional params `flow_id` and `switch_id`. Response example:
+
+~~~
+[ 
+    {
+        "switch_id": "00:00:11:22:33:44:55:66",
+        "flow_id": "string"
```

Comment:
if one switch flows are supported, you need at least port to this api, also, consider one-switch one-port flows as well, maybe it's worth having vlan_vid too

Reply:
All one-switch and one-port flows will have different `flow_id` and the same `switch_id` so we can separate them. Why do we need extra fields like port or vlan? This API should support filtration by switch, flow and both of them.

Reply:
ok, i have one switch flow on sw 1 how I enable loop only for src endpoint of this flow since the switch id is exactly the same?